wrappers/wrapper.rb

```diff
@@ -446,10 +464,74 @@ def assert_not_too_many_visits_in_route(vrp)
       }
     end
 
+    def assert_homogeneous_router_definitions(vrp)
+      vrp.vehicles.group_by{ |vehicle|
+        [vehicle.router_mode, vehicle.dimensions, vehicle.router_options]
+      }.size <= 1
+    end
+
+    def assert_homogeneous_costs(vrp)
+      vrp.vehicles.group_by{ |vehicle|
+        [vehicle.cost_time_multiplier, vehicle.cost_distance_multiplier, vehicle.cost_value_multiplier]
+      }.size <= 1
+    end
+
+    def assert_no_exclusion_cost(vrp)
+      vrp.services.none?(&:exclusion_cost) && vrp.shipments.none?(&:exclusion_cost)
+    end
+
+    def assert_only_time_dimension(vrp)
+      vrp.vehicles.none? { |vehicle|
+        vehicle.cost_distance_multiplier.to_f.positive? ||
+          vehicle.cost_value_multiplier.to_f.positive? ||
+          vehicle.distance
+      }
+    end
+
+    def assert_only_distance_dimension(vrp)
+      vrp.vehicles.none?{ |vehicle|
+        vehicle.cost_time_multiplier.to_f.positive? ||
+          vehicle.cost_value_multiplier.to_f.positive? ||
+          vehicle.duration ||
+          vehicle.timewindow&.end
+      } && vrp.services.none?{ |service| service.activity.timewindows&.any? }
+    end
+
+    def assert_only_value_dimension(vrp)
+      vrp.vehicles.none?{ |vehicle|
+        vehicle.cost_time_multiplier.to_f.positive? ||
+          vehicle.cost_distance_multiplier.to_f.positive? ||
+          vehicle.duration ||
+          vehicle.distance ||
+          vehicle.timewindow&.end
+      }
+    end
+
+    def assert_single_dimension(vrp)
+      vrp.vehicles.empty? || (assert_only_time_dimension(vrp) ^ assert_only_distance_dimension(vrp) ^ assert_only_value_dimension(vrp))
+    end
+
     def assert_no_route_if_schedule_without_periodic_heuristic(vrp)
       vrp.routes.empty? || !vrp.schedule_range_indices || vrp.preprocessing_first_solution_strategy.to_a.include?('periodic')
     end
 
+    # TODO: Need a better way to represent solver preference
+    def assert_small_minimum_duration(vrp)
+      vrp.resolution_minimum_duration.nil? || vrp.vehicles.empty? || vrp.resolution_minimum_duration / vrp.vehicles.size < 5000
```

Comment:
|| vrp.services.empty? && vrp.shipments.empty?  ?

Reply:
`vrp.vehicles.empty?` verification is only present to avoid the division by zero during the solver selection

Reply:
this is only for "duration" The reason why we check vehicles is that we want to calculate "per vehicle" duration (sent by mt-web for example) and if it is not too high we apply Vroom but if the user asked specifically asked +5 seconds per vehicle then we launch or-tools even if it can be solved by Vroom.

Reply:
Adeline your question made me think of another issue we talked this morning during the daily point. 

@braktar can you look at vroom's performance benchmarks and check if it ever passes 30 seconds. If so at what size and for what type of problem?

We might need to add another assert for vroom to prevent it "accepting" too big VRP's

Reply:
Vroom Handke large dose problems. The limitation would probably come from Matrix computation https://github.com/VROOM-Project/vroom/wiki/Benchmarks

Reply:
I agree the matrix is definitely a problem. Still there are instance for which Vroom takes long time as well. For example, for `CVRPLIB` instances the longest time is 4+ minutes (I assume this is the instance with 1000 jobs).